ts/components/error/WorkunitGenericFailure.tsx

```diff
@@ -44,8 +44,8 @@ const WorkunitGenericFailure = (props: Props): React.ReactElement => {
   );
 };
 
-const mapDispatchToProps = (dispatch: Dispatch) => ({
-  cancel: () => dispatch(NavigationActions.back())
+const mapDispatchToProps = (_: Dispatch) => ({
+  cancel: () => navigateBack()
```

Comment:
At this point I suggest we use `navigateBack` straight from the component. 
`mapDispatchToProps` seems an unnecessary abstraction and once we remove it we get rid of the entire `connect` complexity.